components/log-viewer-webui/client/src/pages/IngestPage/Jobs/utils.ts

```diff
@@ -0,0 +1,118 @@
+import dayjs from "dayjs";
+
+import {JobData} from "../Jobs/typings";
+import {QueryJobsItem} from "./sql";
+
+
+/**
+ * Enum of compression job statuses, matching the `CompressionJobStatus` class in
+ * `job_orchestration.scheduler.constants`.
+ *
+ * @enum {CompressionJobStatus}
+ */
+enum CompressionJobStatus {
+    PENDING = 0,
+    RUNNING = 1,
+    SUCCEEDED = 2,
+    FAILED = 3,
+}
+
+/**
+ * Convert JobStatus to Antd's badge string.
+ *
+ * @param status
+ * @return
+ * @throws an Error if status is not a CompressionJobStatus.
+ */
+const convertJobStatusToBadgeString = (status: CompressionJobStatus) => {
+    switch (status) {
+        case CompressionJobStatus.PENDING:
+            return "warning";
+        case CompressionJobStatus.RUNNING:
+            return "processing";
+        case CompressionJobStatus.SUCCEEDED:
+            return "success";
+        case CompressionJobStatus.FAILED:
+            return "error";
+        default:
+            throw new Error("Unexpected status value.");
+    }
+};
+
+const BYTES_PER_KIBIBYTE = 1024;
```

Comment:
Shall we port this https://github.com/y-scope/yscope-log-viewer/blob/main/src/utils/units.ts into `components/log-viewer-webui/client/src/utils`? then we can replace `computeHumanSize`